### PR TITLE
ProjectPasskeys: fix(jans-fido2):  Major FIDO2 / Passkeys upgrade

### PR DESCRIPTION
**Pull request by maduvena**
This PR completely revamps jans-fido2, to enable support for passkeys, and bring the server up to spec.

So far changes:

- Added support for ED25519, RS384, RS512, ES384, ES512 #9086 
- Refactoring of the assertions and attestation requests/responses to Jackson ObjectMapper #9023
- Removed legacy Android Safetynet and Keystore attestations #8901

### CODE REVIEW

File: docs/admin/fido/config.md [hunk 1]
```diff
@@ -26,21 +26,21 @@ tags:
...
+| mdsTocsFolder                           | /etc/jans/conf/fido2/mds/toc                                                   | MDS TOC files folder.                                                                                   |
+| serverMetadataFolder                    | /etc/jans/conf/fido2/server_metadata                                           | Authenticators metadata in json format. Example: virtual devices.                                       |
+| metadataUrlsProvider                    | https://mds3.fido.tools                                                        | String value to provide source of URLs with external metadata.                                          |
+| enabledFidoAlgorithms                   | ["RS256","ES256"]                                                              |                                                                                                         |
+| rp                                      | [{"id":"https://my-jans-server.jans.io","origins":["my-jans-server.jans.io"]}] | Requested party id.                                                                                     |
```
Comment (yackermann) [inline docs/admin/fido/config.md]: id -> rpId, and it should be set to host "my-jans-server.jans.io"

origins need to be ["https://my-jans-server.jans.io"]

File: jans-fido2/model/src/main/java/io/jans/fido2/ctap/CoseEdDSAAlgorithm.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package io.jans.fido2.ctap;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public enum CoseEdDSAAlgorithm {
```
Comment (yackermann) [inline jans-fido2/model/src/main/java/io/jans/fido2/ctap/CoseEdDSAAlgorithm.java]: EDDSA can be dropped. FIDO never supported, and never will. If EDDSA is present then return error.

File: jans-fido2/model/src/main/java/io/jans/fido2/model/assertion/AssertionOptions.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+    public void setUserVerification(UserVerification userVerification) {
+        this.userVerification = userVerification;
+    }
+
+    public String getDocumentDomain() {
+        return documentDomain;
+    }
+
+    public void setDocumentDomain(String documentDomain) {
```
Comment (yackermann) [inline jans-fido2/model/src/main/java/io/jans/fido2/model/assertion/AssertionOptions.java]: what is documentDomain?

File: jans-fido2/model/src/main/java/io/jans/fido2/model/attestation/AttestationOptions.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    public AuthenticatorSelection getAuthenticatorSelection() {
+        return authenticatorSelection;
+    }
+
+    public void setAuthenticatorSelection(AuthenticatorSelection authenticatorSelection) {
+        this.authenticatorSelection = authenticatorSelection;
+    }
+
+    public String getDocumentDomain() {
+        return documentDomain;
```
Comment (yackermann) [inline jans-fido2/model/src/main/java/io/jans/fido2/model/attestation/AttestationOptions.java]: Same. What is a document domain?

File: jans-fido2/model/src/main/java/io/jans/fido2/model/common/PublicKeyCredentialDescriptor.java [hunk 5]
```diff
@@ -20,16 +23,16 @@
 public class PublicKeyCredentialDescriptor {
 
     private String type;
-    private String transports[];
+    private String[] transports;
```
Comment (yackermann) [inline jans-fido2/model/src/main/java/io/jans/fido2/model/common/PublicKeyCredentialDescriptor.java]: Transports need typisation.

File: jans-fido2/server/src/main/java/io/jans/fido2/service/operation/AssertionService.java [hunk 6]
```diff
@@ -114,107 +117,107 @@ public class AssertionService {
...
 
 		Fido2AuthenticationData entity = new Fido2AuthenticationData();
 		entity.setUsername(username);
 		entity.setChallenge(challenge);
 		entity.setDomain(documentDomain);
 		entity.setUserVerificationOption(userVerification);
 		entity.setStatus(Fido2AuthenticationStatus.pending);
-		if (params.hasNonNull(CommonVerifiers.SUPER_GLUU_APP_ID)) {
-			entity.setApplicationId(params.get(CommonVerifiers.SUPER_GLUU_APP_ID).asText());
+		if (!Strings.isNullOrEmpty(assertionOptions.getSuperGluuAppId())) {
```
Comment (yackermann) [inline jans-fido2/server/src/main/java/io/jans/fido2/service/operation/AssertionService.java]: SuperGluu needs to be removed

File: jans-fido2/server/src/main/java/io/jans/fido2/service/operation/AttestationService.java [hunk 7]
```diff
@@ -210,31 +219,31 @@ public ObjectNode options(JsonNode params) {
...
+		commonVerifiers.verifyAssertionType(attestationResult.getType());
 
 		// Get response
-		JsonNode responseNode = params.get("response");
+		//JsonNode responseNode = params.get("response");
 
 		// Verify client data
-		JsonNode clientDataJSONNode = commonVerifiers.verifyClientJSON(responseNode);
+		JsonNode clientDataJSONNode = commonVerifiers.verifyClientJSON(attestationResult.getResponse().getClientDataJSON());
 		if (!superGluu) {
```
Comment (yackermann) [inline jans-fido2/server/src/main/java/io/jans/fido2/service/operation/AttestationService.java]: SuperGluu needs to be nuked

File: jans-fido2/server/src/main/java/io/jans/fido2/service/sg/converter/AssertionSuperGluuController.java [hunk 8]
```diff
@@ -132,20 +133,19 @@ public ObjectNode buildFido2AssertionStartResponse(String userName, String keyHa
...
-
-        params.put("username", userName);
-        params.put("session_id", sessionId);
-
-        log.debug("Prepared U2F_V2 assertions options request: {}", params);
-        return params;
+        AssertionOptions assertionOptions = new AssertionOptions();
+        // Add all required parameters from request to allow process U2F request
+        assertionOptions.setSuperGluuRequest(true);
+        assertionOptions.setSuperGluuAppId(appId);
```
Comment (yackermann) [inline jans-fido2/server/src/main/java/io/jans/fido2/service/sg/converter/AssertionSuperGluuController.java]: Super gluu nuuuuked

**Review by yackermann — CHANGES_REQUESTED** [review]

File: jans-fido2/model/src/main/java/io/jans/fido2/model/attestation/AttestationOptions.java (hunk 4 above)
Comment (maduvena) [reply jans-fido2/model/src/main/java/io/jans/fido2/model/attestation/AttestationOptions.java]: Origin

File: jans-fido2/server/src/main/java/io/jans/fido2/service/operation/AttestationService.java (hunk 7 above)
Comment (maduvena) [reply jans-fido2/server/src/main/java/io/jans/fido2/service/operation/AttestationService.java]: Done in #9453

File: jans-fido2/server/src/main/java/io/jans/fido2/service/operation/AssertionService.java (hunk 6 above)
Comment (maduvena) [reply jans-fido2/server/src/main/java/io/jans/fido2/service/operation/AssertionService.java]: #9453 done

File: jans-fido2/model/src/main/java/io/jans/fido2/model/attestation/AttestationOptions.java (hunk 4 above)
Comment (maduvena) [reply jans-fido2/model/src/main/java/io/jans/fido2/model/attestation/AttestationOptions.java]: changed in #8906

**Comment by yackermann** [comment]
Thanks @imran-ishaq @moabu 

I will review changes tomorrow

**Comment by moabu** [comment]
Closed in favor of a rebase here https://github.com/JanssenProject/jans/pull/10078

**Pull request closed without merging**